valohai_yaml/pipelines/conversion.py

```diff
@@ -91,6 +91,9 @@ def convert_executionlike_node(
                 step_data.pop("no-output-timeout"),
             )
 
+        for key, input in step.inputs.items():
+            step.inputs[key] = input.with_edge_merge_mode_applied()
```

Comment:
1. Should this occur before `step.serialize()` above?
2. Now the `step` is mutated 😁 IOW, if there is a pipeline that has multiple nodes that use the same step, the `convert_executionlike_node()` calls may use a slightly different step. If I'm reasoning about this right, it shouldn't make a difference here, but it's still not good form. Consider:
   ```python
   config = Config()
   print(config.nodes["train"].default)  # ["foo", "bar", "baz"]
   p = PipelineConverter(config, "foo")
   x = p.convert_pipeline(config.pipelines["pipe"])
   print(config.nodes["train"].default)  # []
   ```